pkg/compact/compact.go

```diff
@@ -234,7 +244,24 @@ func (s *Syncer) GarbageCollect(ctx context.Context) error {
 
 	begin := time.Now()
 
-	garbageIds := s.duplicateBlocksFilter.DuplicateIDs()
+	duplicateIds := s.duplicateBlocksFilter.DuplicateIDs()
+	deletionMarkIDs := s.ignoreDeletionMarkFilter.DeletionMarkBlocks(0)
+
+	// GarbageIds contains the duplicateIds, since these blocks can be replaced with other blocks.
+	// We also remove ids present in deletionMarkIDs since these blocks are already marked for deletion.
+	garbageIds := []ulid.ULID{}
+	for _, id := range duplicateIds {
```

Comment:
`ID` should be `ID` never `Id` so: 
```suggestion
	for _, id := range duplicateIDs {
```

Reply:
I will take a note of this and avoid it in next commits.